onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -163,35 +164,63 @@ def create_attention_node(self, mask_index: str, q_matmul: NodeProto, k_matmul:
         kw = NumpyHelper.to_array(k_weight)
         vw = NumpyHelper.to_array(v_weight)
 
-        # Check if all matrices have the same shape
-        assert qw.shape == kw.shape == vw.shape
+        # Check if all matrices have the same shape or q and k have same shape
+        assert qw.shape == kw.shape == vw.shape or qw.shape == kw.shape
```

Comment:
"assert qw.shape == kw.shape == vw.shape or qw.shape == kw.shape" can be simplified as "assert qw.shape == kw.shape"